packages/compat/src/compat-app-builder.ts

```diff
@@ -350,7 +351,24 @@ export class CompatAppBuilder {
       result.unshift({
         kind: 'in-memory',
         relativePath: '_ember_env_.js',
-        source: `window.EmberENV={ ...(window.EmberENV || {}), ...${JSON.stringify(emberENV, null, 2)} };`,
+        source: `
+        (function() {
+          function merge(e, t) {
+              for (var r in t)
+                  e[r] = t[r]
+              return e
+          }
+          try {
+            var metaName = '${appName}' + '/config/environment';
+            var rawConfig = document.querySelector('meta[name="' + metaName + '"]').getAttribute('content');
+            var config = JSON.parse(decodeURIComponent(rawConfig));
+            window.EmberENV = merge(window.EmberENV || {}, config.EmberENV);
```

Comment:
need to make sure that this propagates variables already existing on EmberENV especially in the catch clause for ember Inspector

we most likely want to leave EmberENV intact instead of overriding it in case it errors